docs/Makefile

```diff
@@ -2,9 +2,7 @@
 #
 
 # You can set these variables from the command line.
-# Treat warnings as errors.... hierarchy.rst is just a dump of notes ATM
-# SPHINXOPTS    = -W
-SPHINXOPTS    =
+SPHINXOPTS    = -W
```

Comment:
are we really ready for this? ;)

Reply:
Yes it passed without errors.

The "discussion" docs have been moved to markdown in design/ which were probably causing most of the problems. I only had to make minimal changes to avoid.

The weird thing was that without this the doc build passed if it completed with errors (seems like it this should only affect warnings?) I guess non-fatal errors were treated as warnings?